src/zkp/piprm.rs

```diff
@@ -98,20 +96,20 @@ fn generate_challenge_bytes(
     Ok(challenges.into())
 }
 
-impl Proof for PiPrmProof {
-    type CommonInput = RingPedersen;
-    type ProverSecret = PiPrmSecret;
+impl Proof2 for PiPrmProof {
+    type CommonInput<'a> = &'a RingPedersen;
```

Comment:
We ofen have "wrapper" structs for the `CommonInput`. PiPRM does not, so we just have a `&'a RingPedersen` here.